### DISCUSSION TITLE
Roles without "Admin Access" cannot see messages of errors thrown by SQL trigger

### DISCUSSION THREAD

**Discussion by powerseed** [directus/directus#8127] [Feature Requests]
Hi all~

I have a trigger in my MySQL database that checks data integrity when inserting and updating.

If the check doesn't pass, an error with message will be thrown using `SIGNAL SQLSTATE`.

The potential bug is about the error message. 

When I use an `Admin` role, I can see the error message. A dialog pops up in Directus which shows info like below.
```
[INTERNAL_SERVER_ERROR] {{ error message configured in the SQL trigger }}
```

However, if I use a role without `Admin Access`, The dialog still pops up, but the info is just like
```
[INTERNAL_SERVER_ERROR] An unexpected error occurred.
```
and I cannot see the detailed error message configured in the SQL trigger.

I would like to see this error message. Could anyone teach me how to make it show up when using non-Admin role?

Thanks in advance!

### To Reproduce

1. Add a trigger to a SQL schema, which should throw errors using `SIGNAL SQLSTATE` statement under certain circumstance. This error should contain a user-configured message. 
2. Log in Directus GUI using a role without `Admin Access`. 
3. In Directus GUI, conduct CRUD operations on the schema mentioned in `#1` and meet the condition that throws an error.
4. A dialog should show up in Directus GUI saying `[INTERNAL_SERVER_ERROR] An unexpected error occurred.`, meaning the user-configured message mentioned in `#1` is missing, which it should not.

**Comment by rijkvanzanten** [comment]
Heya! Database errors that aren't abstracted away by the Directus API itself aren't shown to regular users for security reasons. We don't want to leak any potential sensitive information about the database you're using based on the raw error message coming from the database. 

The best way forwards would be to add an exception abstraction for triggers in the various databases we support, so we can have a dedicated "TRIGGER_WARNING" (name tbd) error in the API that can safely return the error message

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
